airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -181,92 +162,27 @@ public void run(final ConnectionUpdaterInput connectionUpdaterInput) throws Retr
 
-      final GeneratedJobInput jobInputs = getJobInput();
-
-      reportJobStarting();
-      StandardSyncOutput standardSyncOutput = null;
-
-      try {
-        final SyncCheckConnectionFailure syncCheckConnectionFailure = checkConnections(jobInputs);
-        if (syncCheckConnectionFailure.isFailed()) {
-          final StandardSyncOutput checkFailureOutput = syncCheckConnectionFailure.buildFailureOutput();
-          workflowState.setFailed(getFailStatus(checkFailureOutput));
-          reportFailure(connectionUpdaterInput, checkFailureOutput, FailureCause.CONNECTION);
-        } else {
-          standardSyncOutput = runChildWorkflow(jobInputs);
-          workflowState.setFailed(getFailStatus(standardSyncOutput));
-
-          if (workflowState.isFailed()) {
-            reportFailure(connectionUpdaterInput, standardSyncOutput, FailureCause.UNKNOWN);
-          } else {
-            reportSuccess(connectionUpdaterInput, standardSyncOutput);
-          }
-        }
+      final GeneratedJobInput jobInputs = generateJobInput(connectionUpdaterInput);
 
-        prepareForNextRunAndContinueAsNew(connectionUpdaterInput);
-      } catch (final ChildWorkflowFailure childWorkflowFailure) {
-        // when we cancel a method, we call the cancel method of the cancellation scope. This will throw an
-        // exception since we expect it, we just
-        // silently ignore it.
-        if (childWorkflowFailure.getCause() instanceof CanceledFailure) {
-          // do nothing, cancellation handled by cancellationScope
-          recordWorkflowFailureCountMetric(connectionUpdaterInput, FailureCause.CANCELED);
-        } else if (childWorkflowFailure.getCause()instanceof final ActivityFailure af) {
-          // Allows us to classify unhandled failures from the sync workflow. e.g. If the normalization
-          // activity throws an exception, for
-          // example, this lets us set the failureOrigin to normalization.
-          workflowInternalState.getFailures().add(FailureHelper.failureReasonFromWorkflowAndActivity(
-              childWorkflowFailure.getWorkflowType(),
-              af.getActivityType(),
-              af.getCause(),
-              workflowInternalState.getJobId(),
-              workflowInternalState.getAttemptNumber()));
-          reportFailure(connectionUpdaterInput, standardSyncOutput, FailureCause.ACTIVITY);
-          prepareForNextRunAndContinueAsNew(connectionUpdaterInput);
-        } else {
-          workflowInternalState.getFailures().add(
-              FailureHelper.unknownOriginFailure(childWorkflowFailure.getCause(), workflowInternalState.getJobId(),
-                  workflowInternalState.getAttemptNumber()));
-          reportFailure(connectionUpdaterInput, standardSyncOutput, FailureCause.WORKFLOW);
-          prepareForNextRunAndContinueAsNew(connectionUpdaterInput);
-        }
-      }
+      checkConnectionAndRunSync(connectionUpdaterInput, jobInputs);
```

Comment:
Looks like the interesting part is actually `checkConnection` which I would rename to `checkConnectionAndRunSync`.

What do you think of the following split instead of calling `checkConnectionAndRunSync`:
```
reportJobStarting();
try {
  final SyncCheckConnectionFailure syncCheckConnectionFailure = checkConnectionAndRunSync();
  if (syncCheckConnectionFailure.isFailed()) {
    ...
  } else {
    ...
  }
} catch (final ChildWorkflowFailure childWorkflowFailure) {
  handleSyncFailure(connectionUpdaterInput, childWorkflowFailure);
}
```

Reply:
Done